src/scene.lisp

```diff
@@ -141,3 +150,32 @@
         (:later `(unload-scene-all-later ,scene))
         (:never nil)
         (t (error ":FREE must be :NOW, :LATER, or :NEVER")))))
+
+(defparameter *scene* nil) ; the current scene in the game loop
+
+;; TODO Perhaps make this generic so users can define :before, :after, :around methods?
+(defun switch-scene (new-scene)
+  "Switch to NEW-SCENE, loading it & unloading the previous scene."
+  (unless (eq new-scene *scene*)
+    (when *scene* (tear-down-scene *scene*))
+    (setf *scene* new-scene)
+    (setup-scene *scene*)))
```

Comment:
On second thought, this should definitely be generic for the same reason setup-scene/tear-down-scene are. User overriding and defining methods on their own scene types inheriting from game-scene.

Reply:
Done https://github.com/defun-games/claylib/pull/18/commits/b7c783ae7d9ab0bafde371de21350d1933c7a636